src/Installer/DNS/Linux/DnsDock.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Dock\Installer\DNS\Linux;
+
+use Dock\Installer\InstallContext;
+use Dock\Installer\InstallerTask;
+use SRIO\ChainOfResponsibility\DependentChainProcessInterface;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+use Symfony\Component\Process\Process;
+
+class DnsDock extends InstallerTask implements DependentChainProcessInterface
+{
+    private $processRunner;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function dependsOn()
+    {
+        return ['docker'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return 'dns';
+    }
+
+    public function run(InstallContext $context)
+    {
+        $this->processRunner = $context->getProcessRunner();
+
+        $userInteraction = $context->getUserInteraction();
+        $userInteraction->writeTitle('Configure DNS resolution for Docker containers');
+
+        if (! $this->hasDockerOpts($processRunner)) {
+            $this->runCommand('echo \'DOCKER_OPTS="--bip=172.17.42.1/24 --dns 172.17.42.1"\' | sudo tee -a /etc/default/docker');
+            $this->runCommand('sudo service docker restart');
+        }
+
+        $this->runCommand('sudo docker start dnsdock || sudo docker run -d -v /var/run/docker.sock:/var/run/docker.sock --name dnsdock -p 172.17.42.1:53:53/udp tonistiigi/dnsdock');
```

Comment:
Why `sudo` for starting docker containers ?

Moreover, you may be missing a [restart policy](https://docs.docker.com/articles/host_integration/) because I think that it will work when running the install script but not after reboot. And it'll also fail when this container is already running I suppose.